code/go/0chain.net/core/util/taskqueue/taskqueue.go

```diff
@@ -0,0 +1,242 @@
+package taskqueue
+
+import (
+	"container/heap"
+	"context"
+	"sync"
+	"time"
+
+	"github.com/0chain/common/core/logging"
+	"go.uber.org/zap"
+)
+
+type Task struct {
+	priority int
+	taskFunc func() error
+	errC     chan error
+	doneC    chan struct{}
+	name     string
+	age      time.Time
+}
+
+// taskExecutor is the global task executor
+var taskExecutor *TaskExecutor
+
+// Init initializes the global task executor
+func Init(ctx context.Context) {
+	taskExecutor = NewTaskExecutor(ctx)
+}
+
+// Execute executes a task
+func Execute(typ TaskType, f func() error) error {
+	errC := make(chan error, 1)
+	taskExecutor.Add(newTask(typ, f, errC))
+	return <-errC
+}
+
+// newTask creates a new task
+func newTask(typ TaskType, f func() error, errC chan error) *Task {
+	return &Task{
+		priority: int(typ),
+		name:     typ.String(),
+		taskFunc: f,
+		errC:     errC,
+		doneC:    make(chan struct{}),
+	}
+}
+
+// PriorityQueue is a priority queue of tasks
+type PriorityQueue []*Task
```

Comment:
we could use heap here, it coud give some performance boost

Reply:
Yes, we are actually using the heap, see code how we add task in https://github.com//0chain/0chain/blob/feature/mpt-cache-pq/code/go/0chain.net/core/util/taskqueue/taskqueue.go#L204